app/serializers/recipe_serializer.rb

```diff
@@ -26,4 +40,15 @@ def price
 
     object.price
   end
+
+  def comments
+    return nil if @instance_options.nil? || @instance_options[:context].nil? || @instance_options[:context][:action].nil?
+
+    if @instance_options.fetch(:context, {})[:action] == 'show'
+      return nil if object.comments.empty?
+      return nil unless object.comments
+
+      object.comments
```

Comment:
return unless @instance_options
```suggestion
    return [] unless @instance_options[:include_comments]
    
    object.comments
```